### PR TITLE
Morteza/movement_animation

### PR DESCRIPTION
**Pull request by morteza-binary**

### CODE REVIEW

**Comment by kelly-binary** [comment]
@morteza-binary LGTM

File: sass/_themes.scss [hunk 1]
```diff
@@ -420,7 +421,8 @@ $themes: (
         HighestLowestMarkerLabelBg: $COLOR_DARK_GRAY_2,
         HighestLowestMarkerLabelText: $COLOR_WHITE,
         HighestSpotColor: $COLOR_GREEN_1,
-        LowestSpotColor: $COLOR_RED
+        LowestSpotColor: $COLOR_RED,
+        ChartBackgroundOpacity: 0.4
```
Comment (brandon-binary) [inline sass/_themes.scss]: The `ChartBackgroundOpacity` for white theme is 0.5, is this suppose to be 0.4?

File: sass/styles/chart.scss [hunk 2]
```diff
@@ -761,6 +761,28 @@ html, body {
...
+        @include themify($themes) {
+            opacity: themed('ChartBackgroundOpacity');
+        }
+        background-image: url('https://static.cdnpub.info/traderoom/map.png'); // TODO the image's url should be changed after getting new image from UI/UX team.
+        background-repeat: no-repeat;
+        background-position: 50% 0;
+        -ms-background-size: cover;
+        -o-background-size: cover;
+        -moz-background-size: cover;
+        -webkit-background-size: cover;
```
Comment (brandon-binary) [inline sass/styles/chart.scss]: just out of curiosity, isn't there a plugin that add all the vendor prefixes for us? Why are we not using it?

File: src/components/ui/Animation.js [hunk 3]
```diff
@@ -323,4 +228,50 @@ export default function animateChart(stx, animationParameters, easeMachine) {
...
+            if (stx.animations.liveScroll.running) {
+                stx.animations.liveScroll.stop();
+            }
+
+            stx.animations.liveScroll.run((bar) => {
+                this.micropixels = -bar;
+                this.draw();
+            }, 0, stx.layout.candleWidth);
+        } else if (!stx.isHome() || stx.chart.lockAutoScroll) {
+            if (stx.animations.liveScroll) {
```
Comment (kelly-binary) [inline src/components/ui/Animation.js]: should this check be `stx.animations.liveScroll.running` ?

File: src/store/ChartStore.js [hunk 4]
```diff
@@ -451,6 +451,7 @@ class ChartStore {
         manageElement.textContent = t.translate('right-click to manage');
 
         if (this.state.isAnimationEnabled) animateChart(stxx, { stayPut: true }, tickAnimator);
+        stxx.chart.lockScroll = true;
```
Comment (kelly-binary) [inline src/store/ChartStore.js]: should this be `lockAutoScroll`?

File: src/components/ui/Animation.js [hunk 5]
```diff
@@ -52,18 +60,20 @@ export default function animateChart(stx, animationParameters, easeMachine) {
...
+        stx.chart.granularity = calculateGranularity(this.layout.interval, this.layout.timeUnit);
 
-    function initMarketSessionFlags() {
-        filterSession = false;
-        nextBoundary = null;
-    }
+        if (stx.chart.granularity === 0) {
+            stx.chart.lockScroll = true;
+        } else {
+            stx.chart.lockScroll = false;
```
Comment (kelly-binary) [inline src/components/ui/Animation.js]: ditto, should this be `lockAutoScroll`?

File: src/store/ChartState.js [hunk 6]
```diff
@@ -372,37 +394,33 @@ class ChartState {
...
-                } else {
-                    this.stxx.chart.scroll = scroll;
-                }
-            },
-            0, scrollToTarget);
+            this.stxx.micropixels = 0;
             this.stxx.draw();
+            this.stxx.chart.entryTick = this.stxx.tickFromDate(startEntry.DT); // the calculation of entry tick should be done after draw
+
+            const scrollToTarget = this.stxx.chart.dataSet.length - this.stxx.chart.entryTick;
```
Comment (kelly-binary) [inline src/store/ChartState.js]: this.stxx.scrollTo doesn't seem to work when `scrollToTarget` is 1. 
suggest to keep this a constant of `2` instead :)

File: src/store/ChartState.js [hunk 7]
```diff
@@ -97,14 +97,24 @@ class ChartState {
...
+            this.exportLayout();
+        }
+
+
         if (chartType !== this.chartType && this.context) {
             this.setChartType(chartType);
-        } else if (granularity !== undefined && granularity !== this.granularity) {
+        }
+
+        if (granularity !== undefined && granularity !== this.granularity) {
```
Comment (kelly-binary) [inline src/store/ChartState.js]: any reason changing to `if` instead of `else if`?

File: src/store/ChartStore.js (hunk 4 above)
Comment (morteza-binary) [reply src/store/ChartStore.js]: No, here it should be `lockScroll` to prevent ChartIQ from autoscrolling.

File: sass/_themes.scss (hunk 1 above)
Comment (morteza-binary) [reply sass/_themes.scss]: yes, it's correct to make it more visible in white theme. Anyway this background should be changed after the new one is done by the UI/UX team.

File: sass/styles/chart.scss (hunk 2 above)
Comment (morteza-binary) [reply sass/styles/chart.scss]: I don't think there are any plugins for it in SmartChart project, I should check it again.

File: src/components/ui/Animation.js (hunk 5 above)
Comment (morteza-binary) [reply src/components/ui/Animation.js]: here Also, it should be `lockScroll`

File: src/store/ChartState.js (hunk 7 above)
Comment (morteza-binary) [reply src/store/ChartState.js]: @kelly-binary yes, because it should be possible to change chartType and granularity at the same time.

**Pull request merged**